evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -519,6 +519,22 @@ pub fn apply<DB: DatabaseExt>(
             let account = data.journaled_state.state().get(&h160_to_b160(inner.0)).unwrap();
             abi::encode(&[Token::Uint(account.info.nonce.into())]).into()
         }
+        HEVMCalls::GetNonce0(inner) => {
+            correct_sender_nonce(
+                b160_to_h160(data.env.tx.caller),
+                &mut data.journaled_state,
+                &mut data.db,
+                state,
+            )?;
+
+            // TODO:  this is probably not a good long-term solution since it might mess up the gas
+            // calculations
+            data.journaled_state.load_account(h160_to_b160(inner.0.addr), data.db)?;
+
+            // we can safely unwrap because `load_account` insert inner.0 to DB.
+            let account = data.journaled_state.state().get(&h160_to_b160(inner.0.addr)).unwrap();
+            abi::encode(&[Token::Uint(account.info.nonce.into())]).into()
```

Comment:
Should all be the same as `HEVMCalls::GetNonce1` except `.get(&h160_to_b160(inner.0.addr))`